backend/src/main/java/com/google/blackswan/mock/SimpleAnomalyGenerator.java

```diff
@@ -0,0 +1,142 @@
+
+package com.google.blackswan.mock;
+
+import static java.util.stream.Collectors.toMap;
+
+import com.google.models.*;
+import java.util.Scanner;
+import java.util.*;
+import java.text.ParseException;
+import java.util.logging.Logger;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.lang.Math;
+import java.lang.Object;
+
+/** Generate list of anomalies based on data in the csv file using average and threshold. */
+public class SimpleAnomalyGenerator implements AnomalyGenerator {
+  private static final String DATA_FILE_LOCATION = "/sample-ramen-data.csv";
+  private static final String EXCEPTION_MESSAGE = "Invalid row format.";
+  private static final Logger log = Logger.getLogger(SimpleAnomalyGenerator.class.getName());
+  private static final int THRESHOLD = 13;
+  private static final int NUM_POINTS = 5;
+
+  private List<Anomaly> anomalies;
+  private Map<Timestamp, Integer> data;
+
+  public SimpleAnomalyGenerator() {
+    anomalies = new ArrayList<Anomaly>();
+    data = new LinkedHashMap<>();
+    parseCSV();
+    generateAnomalies();
+  }
+
+  private void generateAnomalies() {
+    int avg = data.values().stream().reduce(0, Integer::sum) / data.size();
+
+    // Find instances where exceed threshold.
+    Map<Timestamp, Integer> anomalyPoints = data.entrySet().stream()
+        .filter(entry -> entry.getValue() - avg > THRESHOLD)
+        .collect(toMap(entry -> entry.getKey(), entry -> entry.getValue()));
+
+    // Create anomaly objects from those instances.
+    for (Map.Entry<Timestamp, Integer> entry : anomalyPoints.entrySet()) {
+      anomalies.add(createAnomalyFromDataPoint(entry.getKey()));
+    }
+  }
+
+  /** TODO: Depending on size of future data, alter algorithm of finding associated data points. */
+  private Anomaly createAnomalyFromDataPoint(Timestamp time) {
+    // TODO: Make const or obtain metric/dimension name from csv file.
+    String metricName = "Interest Over Time";
+    String dimensionName = "Ramen";
+
+    // Convert keys into ArrayList.
+    Set<Timestamp> keys = data.keySet();
+    List<Timestamp> listKeys = new ArrayList<Timestamp>(keys);
+    int indexOfAnomaly = listKeys.indexOf(time); // Potentially slow operation.
+    if (indexOfAnomaly == -1) {
+      // Should never happen since listKeys is converted from keys.
+      throw new AssertionError("Key does not exist in list.");
+    }
+
+    int firstDataPointIndex = (indexOfAnomaly - NUM_POINTS >= 0) ? indexOfAnomaly - NUM_POINTS : 0;
+    int lastDataPointIndex = (indexOfAnomaly + NUM_POINTS < data.size()) ? indexOfAnomaly + NUM_POINTS : data.size() - 1;
+
+    Map<Timestamp, MetricValue> dataPoints = new HashMap<>();
+    for (int k = firstDataPointIndex; k <= lastDataPointIndex; k++) {
+      dataPoints.put(listKeys.get(k), new MetricValue(data.get(listKeys.get(k))));
+    }
+
+    return new Anomaly(time, metricName, dimensionName, dataPoints);
+  }
+
+  private void parseCSV() {
```

Comment:
For future, might be good to consider a separate class for CsvParsing, as it might get shared between different anomaly generators (or other types of code)

Reply:
I will keep this in mind in the future when more classes parses `csv`s!